crates/rustc_codegen_spirv/src/lib.rs

```diff
@@ -284,12 +288,16 @@ impl CodegenBackend for SpirvCodegenBackend {
 
     fn target_override(&self, opts: &config::Options) -> Option<Target> {
         match opts.target_triple {
-            TargetTriple::TargetTriple(ref target_triple) => match &**target_triple {
-                // TODO: Do we want to match *everything* here, since, well, we only support one thing? that will allow
-                // folks to not specify --target spirv-unknown-unknown on the commandline.
-                "spirv-unknown-unknown" => Some(target_options()),
-                _ => None,
-            },
+            TargetTriple::TargetTriple(ref target_triple) => {
+                const ARCH_VENDOR: &str = "spirv-unknown-";
+                if !target_triple.starts_with(ARCH_VENDOR) {
+                    return None;
+                }
+
+                Some(target_options(
+                    target_triple[ARCH_VENDOR.len()..].parse().ok(),
```

Comment:
Should this fail, or produce some diagnostic (warning?) when it fails to parse? seems unfortunate to typo something and have it silently default to `unknown`